aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,469 @@
+            # TODO: @kavitshah logic for averaging april tag pose
+
+        if camera_T_marker is not None:
+            device_T_marker = self.device_T_camera * camera_T_marker
+            ariaWorld_T_device = frame.get("device_pose")
+
+            avg_ariaWorld_T_marker = self.get_running_avg_ariaWorld_T_marker(
+                ariaWorld_T_device=ariaWorld_T_device, device_T_marker=device_T_marker
+            )
+
+            if avg_ariaWorld_T_marker is not None:
+                avg_marker_T_ariaWorld = avg_ariaWorld_T_marker.inverse()
+
+                # Publish marker pose in ariaWorld frame
+                self.static_tf_broadcaster.sendTransform(
+                    generate_TransformStamped_a_T_b_from_spSE3(
+                        avg_marker_T_ariaWorld,
+                        parent_frame="marker",
+                        child_frame="ariaWorld",
+                    )
+                )
+
+            viz_img, outputs = self.april_tag_detector.get_outputs(
+                img_frame=viz_img,
+                outputs=outputs,
+                device_T_marker=device_T_marker,
+                timestamp=frame.get("timestamp"),
+                img_metadata=None,
+            )
+
+        if detect_hand_object:
+            viz_img, object_scores = self.object_detector.process_frame_online(viz_img)
+            if object_label in object_scores.keys():
+                if self.verbose:
+                    plt.imsave(f"frame_{self._in_index}.jpg", viz_img)
+                self.publish_pose_of_interest(frame.get("ros_pose"))
+        self._in_index += 1
+        return viz_img, outputs
+
+    def initialize_april_tag_detector(self, outputs: dict = {}):
+        """
+        Initialize the april tag detector
+        """
+        self.april_tag_detector = AprilTagDetectorWrapper()
+        focal_lengths = self._dst_calib_params.get_focal_lengths()  # type:ignore
+        principal_point = self._dst_calib_params.get_principal_point()  # type: ignore
+        self.april_tag_detector.enable_detector()
+
+        outputs.update(
+            self.april_tag_detector._init_april_tag_detector(
+                focal_lengths=focal_lengths, principal_point=principal_point
+            )
+        )
+        return outputs
+
+    def initialize_object_detector(self, outputs: dict = {}, object_labels: list = []):
+        """
+        Initialize the object in hand detector
+        """
+        self.object_detector = ObjectDetectorWrapper()
+        self.object_detector.enable_detector()
+        meta_objects: List[str] = []
+        outputs.update(
+            self.object_detector._init_object_detector(
+                object_labels + meta_objects,
+                verbose=self.verbose,
+                version=2,
+            )
+        )
+        self.object_detector._core_objects = object_labels
+        self.object_detector._meta_objects = meta_objects
+
+    def get_running_avg_ariaWorld_T_marker(
```

Comment:
should this really be a `AriaLiveStreamer` class function? This seems more generic, no? Maybe move to utils?

Reply:
I also wanted to do so but in this case this function is also maintaining & updating member lists of positions array & quaternion array.

Not sure how to achieve this from an external function to a class. But if we achieve this, then it could be reused for SpotQRDetector too.

Let's discuss this.

Reply:
As discussed, this should be moved to math_utils.py file.

Reply:
moved